extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -71,6 +90,53 @@ class AmpYoutube extends AMP.BaseElement {
     // the playback.
     return false;
   }
+
+  /** @private */
+  buildImagePlaceholder_() {
+    const imgPlaceholder = new Image();
+    const videoid = this.videoid_;
+
+    setStyles(imgPlaceholder, {
+      // Cover matches YouTube Player styling.
+      'object-fit': 'cover',
+      // Hiding the placeholder initially to give the browser time to fix
+      // the object-fit: cover.
+      'visibility': 'hidden'
+    });
+
+    // TODO(mkhatib): Maybe add <source> children to allow the browser to
+    // load the needed size or even better match YTPlayer logic for loading
+    // player thumbnails for different screen sizes for a cache win!
+    imgPlaceholder.src = 'https://i.ytimg.com/vi/' +
+        encodeURIComponent(this.videoid_) + '/sddefault.jpg';
+    imgPlaceholder.setAttribute('placeholder', '');
+    imgPlaceholder.width = this.width_;
+    imgPlaceholder.height = this.height_;
+
+    this.element.appendChild(imgPlaceholder);
+    this.applyFillContent(imgPlaceholder);
+
+    // Because sddefault.jpg isn't available for all videos, we try to load
+    // it and fallback to hqdefault.jpg.
+    loadPromise(imgPlaceholder).then(() => {
+      // A pretty ugly hack since onerror won't fire on YouTube image 404.
+      // This might be due to the fact that YouTube returns data to the request
+      // even when the status is 404. YouTube returns a placeholder image that
+      // is 120x90.
+      if (imgPlaceholder.naturalWidth == 120 &&
+          imgPlaceholder.naturalHeight == 90) {
+        throw new Error('sddefault.jpg is not found');
+      }
+    }).catch(() => {
+      imgPlaceholder.src = 'https://i.ytimg.com/vi/' +
```

Comment:
Yeah I can do that. The cool thing about the current setup is that if the first image failed to load or if YouTube changed their API to return no data then the catch will still work and replace the image. These are speculative so might not really happen.